packages/client-utils/src/types.d.ts

```diff
@@ -34,4 +38,12 @@ export type TypedPublished<T extends string> = T extends keyof PublishedTypeMap
           ? OutcomeRecord
           : T extends `vaultFactory.managers.manager${number}.metrics`
             ? VaultManagerMetrics
-            : unknown;
+            : T extends 'agoricNames.instance'
+              ? Array<[string, Instance]>
+              : T extends 'agoricNames.brand'
+                ? Array<[string, Brand]>
```

Comment:
A lot of the relevant code expects not just a `Brand` but a `Brand<'nat'>` or `Brand<'set'>`.

https://github.com/Agoric/agoric-sdk/blob/b2cb709533024613178948b2b291ce4b9b490932/multichain-testing/test/fast-usdc/fast-usdc.test.ts#L168
https://github.com/Agoric/agoric-sdk/blob/b2cb709533024613178948b2b291ce4b9b490932/multichain-testing/test/fast-usdc/fu-actors.ts#L39

I suppose one can add `as Brand<'nat'>`, but it almost defeats the point. It's also helpful to reassemble the entries into a record. Do you want to add helpers for this sort of thing?

Reply:
agreed this is low value because you'll often want the kind of brand. It's still true tho so I think it's worth having here.